cypress-io/cypress
# fix: Ensure that file watchers are closed before Electron exits

Closes https://github.com/cypress-io/cypress/issues/22026

This PR uses a `will-quit` hook for Electron to ensure that we close all of the DataContext file watchers before the application exits.

See the related issue for more info.

### User facing changelog


* Fixed an issue where file watchers were not completely closed prior to the Cypress app quitting, causing crashes on exit in some macOS environments.

**Testing**:
- Check out develop
- Follow the repro steps in the related issue to reproduce a SIGABORT error upon exiting the application
- Check out this branch
- Follow the steps again and verify that the SIGABORT error will not happen
